### PR TITLE
Add comments to main.rs

### PR DESCRIPTION
**Pull request by YJDoc2**
Start for #14 
I have only commented main.rs, so you can check the commenting style, and let me know if you have something different in mind. If this is fine, after this I'll make a PR of 2-3 files at once as you said in #14.
I have also started a high-level outerview documentation, which will help understand the overall flow and components. I think as I am writing this not knowing much about container runtimes or youki, this will be helpful for new contributors.
Let me know if any changes are required.

### CODE REVIEW

**Review by utam0k — COMMENTED** [review]
@YJDoc2 
Your work is inspiring to me! The comments and documentation will be great for developers, including new contributors. The comments will also greatly increase the likelihood of youki being used as a crate and the comfort of using it.
Have you ever heard of the [broken windows theory](https://www.britannica.com/topic/broken-windows-theory)? The current youki has broken windows due to my failure to comment. However, if you comment, the window will be repaired and cleaned up, and the culture will be such that other developers will also leave comments.
The culture of comments you create will be wonderful.

This is my mistake, but instead of just docker, a high-level container runtime that includes docker might be appropriate.

**Comment by YJDoc2** [comment]
Hey, Thanks!
Can you tell exactly where the changes are required? I kept docker in the flow diagram as it it not final anyways, and in comments I only mentioned docker, as I think currently it is only tested on docker.
Let me know what changes are required and I will update the PR.

File: src/main.rs [hunk 1]
```diff
@@ -66,6 +78,8 @@ impl SubCommand {
     }
 }
 
+/// This is the entry point in the container runtime. The binary is run by docker,
```
Comment (utam0k) [inline src/main.rs:81]: I'd like you to change docker to the high-level container runtime.

File: src/main.rs [hunk 2]
```diff
@@ -1,3 +1,7 @@
+//! # Youki
+//! Container Runtime written in Rust, inspired by [railcar](https://github.com/oracle/railcar)
+//! This crate provides a container runtime which can be used by Docker to run containers.
```
Comment (utam0k) [inline src/main.rs:3]: I'd like you to change docker to the high-level container runtime.

**Review by utam0k — CHANGES_REQUESTED** [review]

**Comment by utam0k** [comment]
@YJDoc2 I apologize for the confusing explanation.
I have commented on it. Basically, if you don't show a concrete example (flow diagram, etc.), I hope you will use the term higher-level container runtime instead of docker.

**Comment by YJDoc2** [comment]
@utam0k 
Hey, I have made the requested changes, please review.

**Review by utam0k — APPROVED** [review]
nice work! I'm looking forward next to your PR.

**Pull request merged**